### ISSUE TITLE
For your consideration: function naming, code inspection

### ISSUE THREAD

**Issue by gigafone** [reagent-project/reagent-utils#4]
Dear Sir,

Upon inspection of your utility library, in particular the namespace session, I have found myself perplexed by your lexicon. My first lamentation is the inconsistent naming in regards to `put!` and `assoc-in!`. I would venture to say that naming the latter `put-in!` would illuminate their familial nature. The second grievance being found in `assoc-in!` where by a lambda is used inside of the `swap!`. Correct me if I speak in ignorance but is `(swap! state #(assoc-in % ks v))` not directly equivalent to the more terse, yet more legible owing to the smaller count of symbols, `(swap! state assoc-in ks v)`?

Good day.

:smile:

**Comment by yogthos** [comment]
Hi,

Most of this is historical as it owed to [lib-noir](https://github.com/noir-clojure/lib-noir/blob/master/src/noir/session.clj) as the code has been moved over as is. With regards to `put!` and `assoc-in!`, I think the naming is consistent with Clojure namings. Note that `put!` accepts a single key,  where `assoc-in!` accepts a path. The `in` part indicates that we're updating a nested structure. This is a similar dynamic to the relationship between `assoc` and `assoc-in` in Clojure core. The name `put!` would've been more consistent as `assoc!`, but as I said that's historic and there's probably not much value in changing the API at this point.

You're absolutely right regarding the `swap!`, I've cleaned that part up. :)
